src/contracts/NFTFactory.ts

```diff
@@ -553,6 +560,16 @@ export class NftFactory extends SmartContractWithAddress {
     if (!dtParams.name || !dtParams.symbol) {
       ;({ name, symbol } = generateDtName())
     }
+
+    let feeTokenDecimals = 18
+    if (dtParams.feeToken !== ZERO_ADDRESS) {
+      try {
+        feeTokenDecimals = await getTokenDecimals(this.signer, dtParams.feeToken)
+      } catch (error) {
+        LoggerInstance.error('getTokenDecimals error', error)
+      }
+    }
```

Comment:
I think it's worth keeping the `if (dtParams.feeToken !== ZERO_ADDRESS)` so that the decimal request can be skipped. If someone wants to use the `ZERO_ADDRESS` it would be helpful if the function doesn't fail (they may use that if they aren't setting any fee). We also use `ZERO_ADDRESS` as the feeToken address in a lot of the tests so they will fail if we remove this check.

Reply:
Although I see it still returns 18 decimals even if `getTokenDecimals` fails so the tests won't fail, they will just log some errors that don't matter. Either way is fine

Reply:
yep sounds good, lets stick with current impl and merge it

Reply:
ok cool